drivers/include/periph/flash.h

```diff
@@ -0,0 +1,207 @@
+#elif FLASH_WRITE_ALIGN == 4
+typedef uint32_t flash_data_word_t;
+#elif FLASH_WRITE_ALIGN == 8
+typedef uint64_t flash_data_word_t;
+#endif
+
+/**
+ * @brief Initialize flash memory access
+ *
+ * If needed flash memory initialization can be performed.
+ * In native mode a file is loaded into memory to simulate
+ * flash.
+ *
+ * @return	Error Code
+ */
+uint8_t flash_init(void);
+
+/**
+ * @brief Translates an address to page number
+ *
+ * This function is needed to calculate the page number
+ * for erase operation
+ *
+ * @param[in]  address Pointer/Memory address
+ * @param[out] page_offset  Address location in page or NULL
+ *
+ * @return             Flash page number
+ */
+flash_page_number_t flash_get_page_number(void *address, flash_page_size_t *page_offset);
+
+/**
+ * @brief Translates an page number to pointer
+ *
+ * This function is needed to calculate the memory address
+ * of an flash page
+ *
+ * @param[in] page     Page number
+ *
+ * @return             Pointer to page or NULL
+ */
+void *flash_get_address(flash_page_number_t page);
+
+/**
+ * @brief Write data to flash
+ *
+ * This function works like memcpy. It can be used to write
+ * a part of a flash page or an single flash page or multiple
+ * flash pages at once. Written data are reread to check if
+ * data are stored correctly. Optionally a written word can
+ * modified to set only changed bits to reach more write cycles.
+ *
+ * The return code should by checked to handle errors.
+ *
+ * @param[in] dest   Address of flash memory page
+ * @param[in] src    Address of source data
+ * @param[in] n      Number of bytes to write
+ *
+ * @return    FLASH_ERROR_SUCCESS
+ * @return    FLASH_ERROR_BROWNOUT
+ * @return    FLASH_ERROR_FB_CONFIG
+ * @return    FLASH_ERROR_LOCKED
+ * @return    FLASH_ERROR_TIMEOUT
+ * @return    FLASH_ERROR_ALIGNMENT
+ * @return    FLASH_ERROR_VERIFY
+ * @return    FLASH_ERROR_ADDR_RANGE
+ */
+uint8_t flash_memcpy(void *dest, const void *src, size_t n);
+
+/**
+ * @brief Write data to flash without checks
+ *
+ * This function works like memcpy. It can be used to write
+ * a part of a flash page or an single flash page or multiple
+ * flash pages at once. There are no alignment, address or correct
+ * write checks, so write to protected flash regions or unaligned
+ * writes are possible. To support all plattforms do an alignment
+ * and address check by your application
+ * This function is used by tests/periph_flash to check if
+ * alignment is configurd correctly
+ *
+ * The return code should by checked to handle errors.
+ *
+ * @param[in] dest   Address of flash memory page
+ * @param[in] src    Address of source data
+ * @param[in] n      Number of bytes to write
+ *
+ * @return    FLASH_ERROR_SUCCESS
+ * @return    FLASH_ERROR_BROWNOUT
+ * @return    FLASH_ERROR_FB_CONFIG
+ * @return    FLASH_ERROR_LOCKED
+ * @return    FLASH_ERROR_TIMEOUT
+ */
+uint8_t flash_memcpy_fast(void *dest, const void *src, size_t n);
```

Comment:
I think we can remove this functionality. I have planned it to check alignment in test suite and for additional functionality. But while i implement an NRF51 version for flash i see that unaligned writes are impossible because i must write an word and wait for finishing opation to proceed with the next word.